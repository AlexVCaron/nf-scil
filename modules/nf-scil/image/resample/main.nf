process IMAGE_RESAMPLE {
    tag "$meta.id"
    label 'process_single'

    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://scil.usherbrooke.ca/containers/scilus_1.6.0.sif':
        'scilus/scilus:1.6.0' }"

    input:
    tuple val(meta), path(image)

    output:
    tuple val(meta), path("*_resampled.nii.gz"), emit: image
    path "versions.yml"           , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def prefix = task.ext.prefix ?: "${meta.id}"
    def ref = task.ext.ref ? "--ref" + task.ext.ref : ""
    def voxel_size = task.ext.voxel_size ? "--voxel_size " + task.ext.voxel_size : ""
    def volume_size = task.ext.volume_size ? "--volume_size " + task.ext.volume_size : ""
    def iso_min = task.ext.iso_min ? "--iso_min" + task.ext.iso_min : ""
    def interp = task.ext.interp ? "--interp " + task.ext.interp : ""
    def f = task.ext.f ? "--f" + task.ext.f : ""
    def enforce_dimensions = task.ext.enforce_dimensions ? "--enforce_dimensions" + task.ext.enforce_dimensions : ""

    def resampling_method = task.ext.voxel_size ? "--voxel_size " + task.ext.voxel_size : (task.ext.volume_size ? "--volume_size " + task.ext.volume_size : "")

    """
    export ITK_GLOBAL_DEFAULT_NUMBER_OF_THREADS=1
    export OMP_NUM_THREADS=1
    export OPENBLAS_NUM_THREADS=1

    scil_resample_volume.py $resampling_method \
        $ref $iso_min $f $enforce_dimensions \
        $interp \
        $image ${prefix}__resampled.nii.gz \

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: 1.6.0
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    scil_resample_volume.py -h

    touch ${prefix}__resampled.nii.gz

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        scilpy: 1.6.0
    END_VERSIONS
    """
}
