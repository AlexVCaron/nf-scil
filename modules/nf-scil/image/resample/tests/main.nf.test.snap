{
    "image - resample - isomin": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__resampled.nii.gz:md5,f3164907d16ceabf606a03ea0efd3a2f"
                    ]
                ],
                "1": [
                    "versions.yml:md5,941ed1ae50ac0d1e646e6dd4910a3593"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__resampled.nii.gz:md5,f3164907d16ceabf606a03ea0efd3a2f"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,941ed1ae50ac0d1e646e6dd4910a3593"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-30T12:12:58.44014"
    },
    "image - resample - nn": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__resampled.nii.gz:md5,8d3e1c9552673b1ce60bc4d090198564"
                    ]
                ],
                "1": [
                    "versions.yml:md5,941ed1ae50ac0d1e646e6dd4910a3593"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__resampled.nii.gz:md5,8d3e1c9552673b1ce60bc4d090198564"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,941ed1ae50ac0d1e646e6dd4910a3593"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-30T12:13:29.068375"
    },
    "image - resample - volsize": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__resampled.nii.gz:md5,7c83b666f9fb2d68643a20d1c7ac21d8"
                    ]
                ],
                "1": [
                    "versions.yml:md5,941ed1ae50ac0d1e646e6dd4910a3593"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__resampled.nii.gz:md5,7c83b666f9fb2d68643a20d1c7ac21d8"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,941ed1ae50ac0d1e646e6dd4910a3593"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-30T12:12:42.889135"
    },
    "image - resample - voxsize": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__resampled.nii.gz:md5,6ffdabab138186db2dab6f98ea6e9ffb"
                    ]
                ],
                "1": [
                    "versions.yml:md5,941ed1ae50ac0d1e646e6dd4910a3593"
                ],
                "image": [
                    [
                        {
                            "id": "test",
                            "single_end": false
                        },
                        "test__resampled.nii.gz:md5,6ffdabab138186db2dab6f98ea6e9ffb"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,941ed1ae50ac0d1e646e6dd4910a3593"
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-30T12:12:27.368376"
    },
    "image - resample - ref": {
        "content": [
            {
                "0": [
                    
                ],
                "1": [
                    
                ],
                "image": [
                    
                ],
                "versions": [
                    
                ]
            }
        ],
        "meta": {
            "nf-test": "0.8.4",
            "nextflow": "23.10.1"
        },
        "timestamp": "2024-04-30T12:13:13.510866"
    }
}